src/main/java/ui/components/FilterTextField.java

```diff
@@ -55,12 +55,19 @@ private void setup() {
         validationSupport.registerValidator(this, (c, newValue) -> {
             boolean wasError = false;
             try {
-                Parser.parse(getText());
-                setStyleForValidFilter();
+                if (!Parser.check(getText())) {
+                    wasError = true;
+                }
```

Comment:
Changed the name. I thought the interface would be nicer. This way the exception is kept within the parser, and callers have to worry about less (syntax is either correct or not).